internal/pkg/deploy/cloudformation/pipeline.go

```diff
@@ -63,3 +66,15 @@ func (cf CloudFormation) UpdatePipeline(in *deploy.CreatePipelineInput) error {
 			StackName: aws.String(pipelineConfig.StackName()),
 		}, cf.waiters...)
 }
+
+func (cf CloudFormation) DeletePipeline(stackName string) error {
+	// Check if the stack exists
+	out, err := cf.describeStack(&cloudformation.DescribeStacksInput{
```

Comment:
Same idempotency caveat here. If CFN stack has been deleted.

Reply:
I think what @iamhopaul123 means is that we can actually remove the `describeStack` call and call directly `cf.delete(stackName)` which handles ignoring the `stackDoesNotExist` error